internal/controller/nnf_storage_controller.go

```diff
@@ -291,13 +316,149 @@ func (r *NnfStorageReconciler) removePersistentStorageReference(ctx context.Cont
+		nnfNodeBlockStorage := &nnfv1alpha1.NnfNodeBlockStorage{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      nnfNodeStorageName(nnfStorage, allocationSetIndex, i),
+				Namespace: node.Name,
+			},
+		}
+
+		result, err := ctrl.CreateOrUpdate(ctx, r.Client, nnfNodeBlockStorage,
+			func() error {
+				dwsv1alpha2.InheritParentLabels(nnfNodeBlockStorage, nnfStorage)
+				dwsv1alpha2.AddOwnerLabels(nnfNodeBlockStorage, nnfStorage)
+
+				labels := nnfNodeBlockStorage.GetLabels()
+				labels[nnfv1alpha1.AllocationSetLabel] = allocationSet.Name
+				nnfNodeBlockStorage.SetLabels(labels)
+
+				if len(nnfNodeBlockStorage.Spec.Allocations) == 0 {
+					nnfNodeBlockStorage.Spec.Allocations = make([]nnfv1alpha1.NnfNodeBlockStorageAllocationSpec, node.Count)
+				}
+
+				if len(nnfNodeBlockStorage.Spec.Allocations) != node.Count {
+					return dwsv1alpha2.NewResourceError("block storage allocation count incorrect. found %v, expected %v", len(nnfNodeBlockStorage.Spec.Allocations), node.Count).WithFatal()
+				}
+
+				for i := range nnfNodeBlockStorage.Spec.Allocations {
+					nnfNodeBlockStorage.Spec.Allocations[i].Capacity = allocationSet.Capacity
+					if len(nnfNodeBlockStorage.Spec.Allocations[i].Access) == 0 {
+						nnfNodeBlockStorage.Spec.Allocations[i].Access = append(nnfNodeBlockStorage.Spec.Allocations[i].Access, node.Name)
+					}
+				}
+
+				return nil
+			})
+		if err != nil {
+			if !apierrors.IsConflict(err) {
+				return nil, err
+			}
+
+			return &ctrl.Result{Requeue: true}, nil
+		}
+
+		if result == controllerutil.OperationResultCreated {
+			log.Info("Created NnfNodeBlockStorage", "Name", nnfNodeBlockStorage.Name, "Namespace", nnfNodeBlockStorage.Namespace)
+		} else if result == controllerutil.OperationResultNone {
+			// no change
+		} else {
+			log.Info("Updated NnfNodeBlockStorage", "Name", nnfNodeBlockStorage.Name, "Namespace", nnfNodeBlockStorage.Namespace)
+		}
+	}
+
+	return nil, nil
+}
+
+// Get the status from all the child NnfNodeBlockStorage resources and use them to build the status
+// for the NnfStorage.
+func (r *NnfStorageReconciler) aggregateNodeBlockStorageStatus(ctx context.Context, nnfStorage *nnfv1alpha1.NnfStorage, allocationSetIndex int) (*ctrl.Result, error) {
+	allocationSet := &nnfStorage.Status.AllocationSets[allocationSetIndex]
+	allocationSet.AllocationCount = 0
+
+	nnfNodeBlockStorageList := &nnfv1alpha1.NnfNodeBlockStorageList{}
+	matchLabels := dwsv1alpha2.MatchingOwner(nnfStorage)
+	matchLabels[nnfv1alpha1.AllocationSetLabel] = nnfStorage.Spec.AllocationSets[allocationSetIndex].Name
+
+	listOptions := []client.ListOption{
+		matchLabels,
+	}
+
+	if err := r.List(ctx, nnfNodeBlockStorageList, listOptions...); err != nil {
+		return &ctrl.Result{Requeue: true}, nil
+	}
+
+	// Ensure that we found all the NnfNodeStorage resources we were expecting
+	if len(nnfNodeBlockStorageList.Items) != len(nnfStorage.Spec.AllocationSets[allocationSetIndex].Nodes) {
+		return &ctrl.Result{}, nil
```

Comment:
shouldn't this error or requeue?

Reply:
All of the NnfNodeBlockStorages were created in createNodeBlockStorage(), so it they don't exist here yet it's because the client cache hasn't been updated to include them yet. When the client cache is updated we'll get an event that triggers this reconciler, so no requeue or error is needed. I'll update the comment to explain